presto-hive/src/main/java/io/prestosql/plugin/hive/recordwriters/TextOutputFormatRecordWriter.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.recordwriters;
+
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.Utilities;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.Reporter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Properties;
+
+public class TextOutputFormatRecordWriter
+        implements ExtendedRecordWriter
+{
+    private final FSDataOutputStream fsDataOutputStream;
+    private final OutputStream wrappedOutputStream;
+    private int rowSeparator;
+
+    public TextOutputFormatRecordWriter(Path path, JobConf jobConf, Properties properties, boolean isCompressed) throws IOException
+    {
+        String rowSeparatorString = properties.getProperty(
+                serdeConstants.LINE_DELIM, "\n");
+        try {
+            rowSeparator = Byte.parseByte(rowSeparatorString);
+        }
+        catch (NumberFormatException e) {
+            rowSeparator = rowSeparatorString.charAt(0);
+        }
+        fsDataOutputStream = path.getFileSystem(jobConf).create(path, Reporter.NULL);
+        wrappedOutputStream = Utilities.createCompressedStream(jobConf, fsDataOutputStream, isCompressed);
+    }
+
+    @Override
+    public long getWrittenBytes()
+    {
+        return fsDataOutputStream.getPos();
```

Comment:
This will return the size of the uncompressed data. We actually want to report the compressed size, i.e. exactly is written to the file system. You can use `CountingOutputStream` from Guava to wrap the compressed output stream.

Reply:
We're getting the size from FSDataOutputStream - which reports the exact bytes written to file system. From what I understood, the compression stream is wrapped over the file system stream.

Reply:
Sorry, you’re right, I wasn’t thinking clearly yesterday.